tests/mgbench/benchmark.py

```diff
@@ -144,45 +104,427 @@ def filter_benchmarks(generators, patterns):
+        or len(read) == 0
+        and percentage_distribution[1] != 0
+        or len(update) == 0
+        and percentage_distribution[2] != 0
+        or len(analytical) == 0
+        and percentage_distribution[3] != 0
+    ):
+        raise Exception(
+            "There is a missing query in group (write, read, update or analytical) for given workload distribution."
+        )
+
+    random.seed(config_distribution)
+
+    # Executing mixed workload for each test
+    if num_args == 6:
+        for query, funcname in queries[group]:
+            full_workload = []
+
+            log.info(
+                "Running query in mixed workload:",
+                "{}/{}/{}".format(
+                    group,
+                    query,
+                    funcname,
+                ),
+            )
+            base_query = getattr(dataset, funcname)
+
+            base_query_group = funcname.rsplit("_", 1)[1]
+
+            if percentage_distribution[0] > 0 and base_query_group == "write":
+                print("Skiping the query in same group as load: " + funcname)
+                continue
+            elif percentage_distribution[1] > 0 and base_query_group == "read":
+                print("Skiping the query in same group as load: " + funcname)
+                continue
+            elif percentage_distribution[2] > 0 and base_query_group == "update":
+                print("Skiping the query in same group as load: " + funcname)
+                continue
+            elif percentage_distribution[3] > 0 and base_query_group == "analytical":
+                print("Skiping the query in same group as load: " + funcname)
+                continue
+
+            options = ["w", "r", "u", "a", "q"]
+            function_type = random.choices(
+                population=options, weights=percentage_distribution, k=num_of_queries
+            )
+
+            for t in function_type:
+                # Get the apropropriate functions with same probabilty
+                if t == "w":
+                    funcname = random.choices(write, k=1)[0]
+                    aditional_query = getattr(dataset, funcname)
+                    full_workload.append(aditional_query())
+                elif t == "r":
+                    funcname = random.choices(read, k=1)[0]
+                    aditional_query = getattr(dataset, funcname)
+                    full_workload.append(aditional_query())
+                elif t == "u":
+                    funcname = random.choices(update, k=1)[0]
+                    aditional_query = getattr(dataset, funcname)
+                    full_workload.append(aditional_query())
+                elif t == "a":
+                    funcname = random.choices(analytical, k=1)[0]
+                    aditional_query = getattr(dataset, funcname)
+                    full_workload.append(aditional_query())
+                elif t == "q":
+                    full_workload.append(base_query())
```

Comment:
Also repetitive (and also the `if ... continue` block above), maybe there could be some sort of an abstract definition, e.g., the `Op` (read, write, ...) and `Workload` (a collection of queries), that would simplify/abstract all the logic about generating the queries. NOTE: if you go there, apply the change to all query generation code.

Reply:
This was rewritten, it is not repetitive anymore.